module-rt/tests/multiple.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+extern crate foundry_module_rt as fmoudle_rt;
+extern crate foundry_process_sandbox as fproc_sndbx;
+
+use fmoudle_rt::coordinator_interface::{FoundryModule, PartialRtoConfig, Port};
+use fmoudle_rt::UserModule;
+use fproc_sndbx::execution::executor::{add_function_pool, execute, Context as ExecutorContext, PlainThread};
+use fproc_sndbx::ipc::{generate_random_name, intra::Intra, Ipc};
+use parking_lot::RwLock;
+use rand::prelude::*;
+use rand::seq::SliceRandom;
+use remote_trait_object::raw_exchange::{import_service_from_handle, HandleToExchange, Skeleton};
+use remote_trait_object::{service, Config as RtoConfig, Context as RtoContext, Service, ServiceRef, ServiceToImport};
+use std::sync::Arc;
+
+#[service]
+trait Pizza: Service {}
+
+#[service]
+trait PizzaBox: Service {}
+
+#[service]
+trait PizzaStore: Service {
+    fn create_pizza(&self) -> ServiceRef<dyn Pizza>;
+    fn wrap_pizza_box(&self) -> ServiceRef<dyn PizzaBox>;
+}
+
+struct SimplePizza;
+impl Service for SimplePizza {}
+impl Pizza for SimplePizza {}
+
+struct SimplePizzaBox {
+    _pizzas: Vec<Box<dyn Pizza>>,
+}
+impl Service for SimplePizzaBox {}
+impl PizzaBox for SimplePizzaBox {}
+
+struct SimplePizzaStore {
+    pizza_pool: Arc<RwLock<Vec<Box<dyn Pizza>>>>,
+}
+impl Service for SimplePizzaStore {}
+impl PizzaStore for SimplePizzaStore {
+    fn create_pizza(&self) -> ServiceRef<dyn Pizza> {
+        ServiceRef::create_export(Box::new(SimplePizza) as Box<dyn Pizza>)
+    }
+
+    fn wrap_pizza_box(&self) -> ServiceRef<dyn PizzaBox> {
+        let mut rng = rand::thread_rng();
+        let mut pool = self.pizza_pool.write();
+        let n = pool.len() - rng.gen_range(0, std::cmp::min(5, pool.len()));
+        let pizzas = pool.split_off(n);
+        ServiceRef::create_export(Box::new(SimplePizzaBox {
+            _pizzas: pizzas,
+        }) as Box<dyn PizzaBox>)
+    }
+}
+struct ModuleA {
+    pizza_stores: Vec<Box<dyn PizzaStore>>,
+    pizza_pool: Arc<RwLock<Vec<Box<dyn Pizza>>>>,
+    pizza_boxes: Vec<Box<dyn PizzaBox>>,
```

Comment:
Could you add a comment that what this test wants to do?

Reply:
updated